docker/build-env/Dockerfile

```diff
@@ -0,0 +1,137 @@
+FROM nvidia/cuda:10.1-devel-ubuntu18.04
```

Comment:
This image is good for testing on GPUs (nothing to test yet, but we will need this later...).
Please add a non CUDA image for testing on multicore.

Reply:
I was thinking to build one big build environment container with all prebuilt dependencies in it, and then do a multistage build where a fresh ubuntu image is extended and binaries are copied over from the previous stage. If you decide not to build for GPUs, then you simply don't copy the binaries over. This process is automated (see below)

Reply:
My concern is that libraries may be compiled in a different way if CUDA is found or not (For example HPX).
If possible I would like to run the tests for a multicore build in an environment without traces of CUDA.